### PR TITLE
Add Odyssey clear button to SearchField

### PR DESCRIPTION
**Pull request by jordankoschei-okta**
`SearchField` was still using the browser-native clear button, which doesn't match Odyssey conventions. This PR adds our own clear button as an `endAdornment`, and makes some related changes necessary to making that work:

- `SearchField` value is now controlled via `useState`. This was necessary because otherwise the Clear button would only appear when the input was controlled.
- `SearchField` now has an optional `onClear` function. The clear button will clear the input regardless, but now consumers can hook additional functionality into that button press, enabling things like clearing a list when the search field is cleared.

### CODE REVIEW

File: packages/odyssey-react-mui/src/SearchField.tsx [hunk 1]
```diff
@@ -76,21 +82,58 @@ const SearchField = forwardRef<HTMLInputElement, SearchFieldProps>(
...
   ) => {
+    const [uncontrolledValue, setUncontrolledValue] = useState("");
+
+    const onChange = useCallback(
+      (e) => {
+        setUncontrolledValue(e.target.value);
+        if (onChangeProp) onChangeProp(e);
+      },
+      [onChangeProp]
+    );
```
Comment (jordankoschei-okta) [inline packages/odyssey-react-mui/src/SearchField.tsx:102]: - Change `e` to `event`
- Make the `useCallback` use `<ChangeEventHandler>`

File: packages/odyssey-react-mui/src/SearchField.tsx [hunk 2]
```diff
@@ -76,21 +82,58 @@ const SearchField = forwardRef<HTMLInputElement, SearchFieldProps>(
...
-      value,
+      value: controlledValue,
     },
     ref
   ) => {
+    const [uncontrolledValue, setUncontrolledValue] = useState("");
+
+    const onChange = useCallback(
+      (e) => {
+        setUncontrolledValue(e.target.value);
```
Comment (jordankoschei-okta) [inline packages/odyssey-react-mui/src/SearchField.tsx:98]: Change `target` to `currentTarget`
```suggestion
        setUncontrolledValue(e.currentTarget.value);
```

File: packages/odyssey-react-mui/src/SearchField.tsx [hunk 3]
```diff
@@ -76,21 +82,58 @@ const SearchField = forwardRef<HTMLInputElement, SearchFieldProps>(
...
+        if (onChangeProp) onChangeProp(e);
+      },
+      [onChangeProp]
+    );
+
+    const onClear = useCallback(() => {
+      setUncontrolledValue("");
+      if (onClearProp) onClearProp();
```
Comment (jordankoschei-okta) [inline packages/odyssey-react-mui/src/SearchField.tsx:106]:
```suggestion
      if (onClearProp) { onClearProp(); }
```

Comment (KevinGhadyani-Okta) [reply packages/odyssey-react-mui/src/SearchField.tsx:106]:
```suggestion
      onClearProp?.();
```

File: packages/odyssey-react-mui/src/SearchField.tsx [hunk 4]
```diff
@@ -104,18 +147,21 @@ const SearchField = forwardRef<HTMLInputElement, SearchFieldProps>(
...
         hasInitialFocus,
+        isDisabled,
         onChange,
         onFocus,
         onBlur,
         placeholder,
         ref,
-        value,
+        uncontrolledValue,
       ]
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/SearchField.tsx:165]: We're missing `controlledValue` in this dependency list.

File: packages/odyssey-storybook/src/components/odyssey-mui/SearchField/SearchField.stories.tsx [hunk 5]
```diff
@@ -21,40 +21,118 @@ const storybookMeta: Meta<SearchFieldProps> = {
...
+        },
+      },
     },
   },
   args: {
     label: "Search",
     placeholder: "Search planets",
   },
   decorators: [MuiThemeDecorator],
+  tags: ["autodocs"],
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-storybook/src/components/odyssey-mui/SearchField/SearchField.stories.tsx:135]: What's this tag?

Comment (jordankoschei-okta) [reply packages/odyssey-storybook/src/components/odyssey-mui/SearchField/SearchField.stories.tsx:135]: This is what generates the Docs page without us needing to include an `mdx` file; it generates the page with the default story, controls, and secondary stories.

**Comment by jordankoschei-okta** [comment]
**Note for review with KG:** We had discussed making this controlled only, but it felt better to me with the option to be controlled or uncontrolled. Are you comfortable merging it with both possibilities available?

File: packages/odyssey-react-mui/src/SearchField.tsx [hunk 6]
```diff
@@ -76,21 +82,59 @@ const SearchField = forwardRef<HTMLInputElement, SearchFieldProps>(
...
+
     const renderFieldComponent = useCallback(
       ({ ariaDescribedBy, id }) => (
         <InputBase
           aria-describedby={ariaDescribedBy}
           autoComplete={autoCompleteType}
           /* eslint-disable-next-line jsx-a11y/no-autofocus */
           autoFocus={hasInitialFocus}
+          endAdornment={
+            uncontrolledValue.length > 0 && (
```
Comment (jordankoschei-okta) [inline packages/odyssey-react-mui/src/SearchField.tsx:124]:
```suggestion
            uncontrolledValue && (
```

File: packages/odyssey-react-mui/src/SearchField.tsx [hunk 7]
```diff
@@ -104,18 +148,22 @@ const SearchField = forwardRef<HTMLInputElement, SearchFieldProps>(
             </InputAdornment>
           }
           type="search"
-          value={value}
+          value={
+            controlledValue !== undefined ? controlledValue : uncontrolledValue
```
Comment (jordankoschei-okta) [inline packages/odyssey-react-mui/src/SearchField.tsx:152]:
```suggestion
            controlledValue === undefined ? uncontrolledValue : controlledValue
```

File: packages/odyssey-react-mui/src/SearchField.tsx [hunk 8]
```diff
@@ -104,18 +148,22 @@ const SearchField = forwardRef<HTMLInputElement, SearchFieldProps>(
...
+          }
         />
       ),
       [
         autoCompleteType,
         hasInitialFocus,
+        isDisabled,
```
Comment (jordankoschei-okta) [inline packages/odyssey-react-mui/src/SearchField.tsx:159]:
```suggestion
        isDisabled,
        onClear,
```

**Review by KevinGhadyani-Okta — APPROVED** [review]

**Comment by KevinGhadyani-Okta** [comment]
There are a few suggestions we made on the call that need to be updated before merging **

**Pull request merged**